microsoft/testsuites/asr/cvt.py

```diff
@@ -0,0 +1,275 @@
+            "UBUNTU-14.04-64",
+            "UBUNTU-16.04-64",
+            "OL6-64",
+            "OL7-64",
+        }
+        extension_test = {
+            "SLES11-SP3-64",
+            "SLES11-SP4-64",
+            "OL6-64",
+            "RHEL7-64",
+        }
+        publisher_name = "Microsoft.Azure.SiteRecovery.Test"
+        extension_name = self.get_extension_name(os=os, log=log)
+        if os in extension_publishers:
+            if os in extension_test:
+                extension_name = extension_name + "Test"
+        else:
+            publisher_name = "Microsoft.Azure.SiteRecovery2.Test"
+
+        cur_time = datetime.datetime.now().isoformat() + "Z"
+        settings = {
+            "publicObject": "",
+            "module": "a2a",
+            "timeStamp": cur_time,
+            "commandToExecute": "Install",
+            "taskId": task_id,
+        }
+        extension = node.features[AzureExtension]
+
+        result = extension.create_or_update(
+            name=extension_name,
+            publisher=publisher_name,
+            type_=extension_name,
+            type_handler_version="1.0",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+        )
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+    def run_script(
+        self, node: Node, log: Logger, test_dir: str
+    ) -> ExecutableResult:
+        timer = create_timer()
+        script: CustomScript = node.tools[self._cvt_script]
+        result = script.run(parameters=test_dir, timeout=19800, sudo=True)
+        log.info(f"Script run with param {test_dir} finished within {timer}")
+        return result
+
+    def copy_cvt_logs(
+        self, log: Logger, node: Node, test_dir: Path, log_path: Path
+    ) -> None:
+        find_tool = node.tools[Find]
+        file_list = find_tool.find_files(
+            test_dir,
+            name_pattern="*.log",
+            file_type="f",
+            sudo=True,
+            ignore_not_exist=True,
+        )
+
+        file_list += find_tool.find_files(
+            test_dir,
+            name_pattern="*.txt",
+            file_type="f",
+            sudo=True,
+            ignore_not_exist=True,
+        )
+
+        for file in file_list:
+            log.info(f"Copying file {file} to {log_path}")
+            try:
+                file_name = file.split("/")[-1]
+                node.shell.copy_back(
+                    node.get_pure_path(file),
+                    log_path / f"{file_name}",
+                )
+            except FileNotFoundError:
+                log.error(f"File {file} doesn't exist.")
+
+    def run_cvt_tests(
+        self,
+        log: Logger,
+        node: Node,
+        log_path: Path,
+        variables: Dict[str, Any],
+    ) -> ExecutableResult:
+        cvt_bin = "indskflt_ct"
+        sas_uri = variables.get("cvt_binary_sas_uri", "")
```

Comment:
when cvt_binary_sas_uri is not set, we need skip the test

Reply:
Added raise SkippedException